tests/test_transformed_data.py

```diff
@@ -0,0 +1,128 @@
+    ("beckers_barley_trellis_plot.py", 120, ["year", "site"]),
+    ("beckers_barley_wrapped_facet.py", 120, ["site", "median_yield"]),
+    ("bump_chart.py", 100, ["rank", "yearmonth_date"]),
+    ("comet_chart.py", 120, ["variety", "delta"]),
+    ("connected_scatterplot.py", 55, ["miles", "gas"]),
+    ("diverging_stacked_bar_chart.py", 40, ["value", "percentage_start"]),
+    ("donut_chart.py", 6, ["value_start", "value_end"]),
+    ("gapminder_bubble_plot.py", 187, ["income", "population"]),
+    ("grouped_bar_chart2.py", 9, ["Group", "Value_start"]),
+    ("hexbins.py", 84, ["xFeaturePos", "mean_temp_max"]),
+    ("histogram_heatmap.py", 378, ["bin_maxbins_40_Rotten_Tomatoes_Rating", "__count"]),
+    ("histogram_scatterplot.py", 64, ["bin_maxbins_10_Rotten_Tomatoes_Rating", "__count"]),
+    ("interactive_legend.py", 1708, ["sum_count_start", "series"]),
+    ("iowa_electricity.py", 51, ["net_generation_start", "year"]),
+    ("isotype.py", 37, ["animal", "x"]),
+    ("isotype_grid.py", 100, ["row", "col"]),
+    ("lasagna_plot.py", 492, ["yearmonthdate_date", "sum_price"]),
+    ("layered_area_chart.py", 51, ["source", "net_generation"]),
+    ("layered_bar_chart.py", 51, ["source", "net_generation"]),
+    ("layered_histogram.py", 113, ["bin_maxbins_100_Measurement"]),
+    ("line_chart_with_cumsum.py", 52, ["cumulative_wheat"]),
+    ("line_percent.py", 30, ["sex", "perc"]),
+    ("line_with_log_scale.py", 15, ["year", "sum_people"]),
+    ("multifeature_scatter_plot.py", 150, ["petalWidth", "species"]),
+    ("natural_disasters.py", 686, ["Deaths", "Year"]),
+    ("normalized_stacked_area_chart.py", 51, ["source", "net_generation_start"]),
+    ("normalized_stacked_bar_chart.py", 60, ["site", "sum_yield_start"]),
+    ("parallel_coordinates.py", 600, ["key", "value"]),
+    ("percentage_of_total.py", 5, ["PercentOfTotal", "TotalTime"]),
+    ("pie_chart.py", 6, ["category", "value_start"]),
+    ("pyramid.py", 3, ["category", "value_start"]),
+    ("stacked_bar_chart_sorted_segments.py", 60, ["variety", "site"]),
+    ("stem_and_leaf.py", 100, ["stem", "leaf"]),
+    ("streamgraph.py", 1708, ["series", "sum_count"]),
+    ("top_k_items.py", 10, ["rank", "IMDB_Rating_start"]),
+    ("top_k_letters.py", 9, ["rank", "letters"]),
+    ("top_k_with_others.py", 10, ["ranked_director", "mean_aggregate_gross"]),
+    ("trellis_area_sort_array.py", 492, ["date", "price"]),
+    ("trellis_histogram.py", 20, ["Origin", "__count"]),
+    ("us_population_over_time.py", 38, ["sex", "people_start"]),
+    ("us_population_over_time_facet.py", 285, ["year", "sum_people"]),
+    ("wilkinson-dot-plot.py", 21, ["data", "id"]),
+    ("window_rank.py", 12, ["team", "diff"]),
+])
+# fmt: on
+def test_primitive_chart_examples(filename, rows, cols):
+    source = pkgutil.get_data(examples_methods_syntax.__name__, filename)
+    chart = eval_block(source)
+    df = chart._transformed_data()
+    assert len(df) == rows
```

Comment:
Do you think it makes sense to also check if this dataframe no nulls? `assert df.notnull().all().all()`

Reply:
I don't think so. When the input DataFrame has nulls it's possible for these to be pass through to the transformed data. Vega-Lite usually filters null values for the columns that are used in the chart, but `transformed_data` returns all of the columns, so the unused columns can still have nulls.